storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1374,5 +1393,29 @@ public static RuntimeException wrapInRuntime(Exception e){
             return new RuntimeException(e);
         }
     }
+
+    public static int getAvailablePort(Integer port) {
+        Integer localPort = null;
+        ServerSocket socket = null;
+        try {
+            socket = new ServerSocket(port);
+            localPort = socket.getLocalPort();
+        } catch (IOException exp) {
+            getAvailablePort(0);
```

Comment:
also this should be done only if the input port is non-zero. similar to what is happening in clojure. since there is no control on recursion depth here.